docs/source/format/CDataInterface.rst

```diff
@@ -0,0 +1,925 @@
+      for (i = 0; i < array->n_buffers; ++i) {
+         free((void *) array->buffers[i]);
+      }
+      free(array->buffers);
+      // Mark released
+      array->release = NULL;
+   }
+
+   void export_float32_utf8_array(
+         int64_t nitems,
+         const uint8_t* float32_nulls, const float* float32_data,
+         const uint8_t* utf8_nulls, const int32_t* utf8_offsets, const uint8_t* utf8_data,
+         struct ArrowArray* array) {
+      struct ArrowArray* child;
+
+      //
+      // Initialize parent array
+      //
+      *array = (struct ArrowArray) {
+         // Data description
+         .length = nitems,
+         .offset = 0,
+         .null_count = 0,
+         .n_buffers = 1,
+         .n_children = 2,
+         .dictionary = NULL,
+         // Bookkeeping
+         .release = &release_malloced_array
+      };
+      // Allocate list of parent buffers
+      array->buffers = malloc(sizeof(void*) * array->n_buffers);
+      array->buffers[0] = NULL;  // no nulls, null bitmap can be omitted
+      // Allocate list of children arrays
+      array->children = malloc(sizeof(struct ArrowArray*) * array->n_children);
+
+      //
+      // Initialize child array #0
+      //
+      child = array->children[0] = malloc(sizeof(struct ArrowArray));
+      *child = (struct ArrowArray) {
+         // Data description
+         .length = nitems,
+         .offset = 0,
+         .null_count = -1,
+         .n_buffers = 2,
+         .n_children = 0,
+         .dictionary = NULL,
+         .children = NULL,
+         // Bookkeeping
+         .release = &release_malloced_array
+      };
+      child->buffers = malloc(sizeof(void*) * array->n_buffers);
+      child->buffers[0] = float32_nulls;
+      child->buffers[1] = float32_data;
+
+      //
+      // Initialize child array #1
+      //
+      child = array->children[1] = malloc(sizeof(struct ArrowArray));
+      *child = (struct ArrowArray) {
+         // Data description
+         .length = nitems,
+         .offset = 0,
+         .null_count = -1,
+         .n_buffers = 3,
+         .n_children = 0,
+         .dictionary = NULL,
+         .children = NULL,
+         // Bookkeeping
+         .release = &release_malloced_array
+      };
+      child->buffers = malloc(sizeof(void*) * array->n_buffers);
+      child->buffers[0] = utf8_nulls;
+      child->buffers[1] = utf8_offsets;
+      child->buffers[2] = utf8_data;
+   }
+
+
+Why two distinct structures?
+============================
+
+In many cases, the same type or schema description applies to multiple,
+possibly short, batches of data.  To avoid paying the cost of exporting
+and importing the type description for each batch, the ``ArrowSchema``
+can be passed once, separately, at the beginning of the conversation between
+producer and consumer.
+
+In other cases yet, the data type is fixed by the producer API, and may not
+need to be communicated at all.
+
+However, if a producer is focussed on one-shot exchange of data, it can
```

Comment:
nit: `focussed` -> `focused`

Reply:
Hmm, thank you. I'll make the change.

Reply:
TBF it can be spelled both ways:
> *Both spellings are used depending on the variety of English. According to Wiktionary: The spelling focused is much more common in the US; however, the spelling focussed is sometimes used in the UK and Canada, and is especially common in Australia and New Zealand.*